spec/tapioca/gem/pipeline_spec.rb

```diff
@@ -3847,5 +3848,128 @@ module ViewHelpers::UrlHelper; end
 
       assert_equal(output, compile)
     end
+
+    it "compile RBIs with location from gem source" do
+      add_ruby_file("foo.rb", <<~RB)
+        class Foo
+          extend T::Helpers
+
+          requires_ancestor { ::Helper }
+
+          module Helper
+            def helper_method; end
+          end
+        end
+
+        module Bar
+          extend T::Sig
+
+          # Some documentation
+          sig { void }
+          def bar; end
+
+          # Some documentation
+          def self.bar; end
+
+          BAR = Foo
+
+          # We shouldn't add the documentation when it points to things like `(eval)`
+          eval("def foo1; end")
+          define_method(:foo2) {}
+        end
+
+        class Baz
+          extend T::Helpers
+          abstract!
+        end
+
+        class Quux < T::Struct
+        end
+
+        class String
+          def foo; end
+        end
+
+        class BasicFoo < BasicObject
+          extend ::T::Sig
+
+          sig { void }
+          def foo; end
+        end
+      RB
+
+      output = template(<<~RBI)
+        # source://the-dep-1.1.2/lib/foo.rb:11
+        module Bar
+          # Some documentation
+          #
+          # source://the-dep-1.1.2/lib/foo.rb:16
+          sig { void }
+          def bar; end
+
+          def foo1; end
+
+          # source://the-dep-1.1.2/lib/foo.rb:25
+          def foo2; end
+
+          class << self
+            # Some documentation
+            #
+            # source://the-dep-1.1.2/lib/foo.rb:19
+            def bar; end
+          end
+        end
+
+        # source://the-dep-1.1.2/lib/foo.rb:21
+        Bar::BAR = Foo
+
+        # source://the-dep-1.1.2/lib/foo.rb:40
+        class BasicFoo < ::BasicObject
+          # source://the-dep-1.1.2/lib/foo.rb:44
+          sig { void }
+          def foo; end
+        end
+
+        # @abstract It cannont be directly instantiated. Subclasses must implement the `abstract` methods below.
+        #
+        # source://the-dep-1.1.2/lib/foo.rb:28
+        class Baz
+          abstract!
+
+          # source://sorbet-runtime-0.5.10125/lib/types/private/abstract/declare.rb:37
+          def initialize(*args, **_arg1, &blk); end
+        end
+
+        # source://the-dep-1.1.2/lib/foo.rb:1
+        class Foo; end
+
+        # source://the-dep-1.1.2/lib/foo.rb:6
+        module Foo::Helper
+          # source://the-dep-1.1.2/lib/foo.rb:7
+          def helper_method; end
+        end
+
+        # source://the-dep-1.1.2/lib/foo.rb:33
+        class Quux < ::T::Struct
+          class << self
+            # source://sorbet-runtime-0.5.10125/lib/types/struct.rb:13
```

Comment:
On this note, can we please interpolate the version number of Sorbet runtime here. Otherwise, we will keep having to update this test every time we do a gem bump.